Training/interface/DataLoader_main.h

```diff
@@ -254,6 +254,14 @@ class DataLoader {
               FillCellGrid(tau, tau_i, outerCellGridRef, false);
               ++tau_i;
             }
+            else {
+              if (include_mismatched)
+                ++tau_i;
+            }
+          }
+          else {
+            if (include_mismatched)
+              ++tau_i;
```

Comment:
Maybe, as previously suggested by Konstantin, we can try to do the following:
1) Return n_batches on every iteration (not just n_tau=1) 
2) add "running" n_batches_current variable in `Data class` so that when you do reshape here: https://github.com/cms-tau-pog/TauMLTools/blob/45babb742f1a15f96950e84d60c95acc9b65f7e9/Training/python/DataLoader.py#L160 you reshape by this n_batches_current to be consistent on the last iteration when tau_i < n_batches.

What @kandrosov think about it?

Reply:
Hm, you mean basically to reshape at all the steps by `self.batch_size` except for the last iteration, where the reshape is by `self.n_total_tau - self. n_batches_current*self. batch_size `?

Reply:
For the first version, it is fine. Improvements for computing performance can be done later, once we have the whole chain functional.
However, please avoid have `if(include_mismatched)` in two places. Add it at the end, e.g.:
```c++
if(!tau_is_set && include_mismatched)
  ++tau_i;
```

Reply:
done! 06ae65f